Breaking up monolith deployments (#367)
OctopusDeploy/blog: blog/2020-11/breaking-up-monolith-deployments/index.md

---
title: Break up monolithic deployments
description: Learn how to break up a large process into smaller deployable components
author: shawn.sesna@octopus.com
visibility: private
published: 2021-11-03
metaImage: 
bannerImage: 
tags:
---

The term `monolithic deployment` often carries a negetive connotation, however, monolithic deployments are the natural progression of an application that has grown over time.  In this post, I'll demonstrate breaking apart a monolithic process into smaller deployable components.

## Pitstop
For this post, I chose the [Pitstop](https://github.com/EdwinVW/pitstop) application to use as an example.  The reason I chose this application was that it contained multiple moving parts:
- Web front end
- APIs
- Microservices
- Databases
- References third-party docker containers

> The original version of this application embedded the database and table creation within the C# code itself.  To make a more useful demonstration, I extracted the database activities and placed them into the deployment process.  In addition, the APIs and Web front end were all dockerized.  Again, I modified this so they could be deployed to Azure instead of containers.  The modified version can be found [here](https://github.com/OctopusSamples/PitStop).

Reviewer: ```suggestion
The original version of this application embedded the database and table creation within the C# code itself.  To make a more useful demonstration, I extracted the database activities and placed them into the deployment process.  In addition, the APIs and Web front end were all dockerized.  Again, I modified this so they could be deployed to Azure instead of containers.  Here's the [modified version](https://github.com/OctopusSamples/PitStop).
```

---
title: Breaking up monolithic deployments
description: Learn how to break up a large process into smaller deployable components
author: shawn.sesna@octopus.com
visibility: private

> published: 2021-11-03

Reviewer: ```suggestion
published: 2020-12-08
```

metaImage: 
bannerImage: 
tags:

> 	- DevOps

Reviewer: ```suggestion
  - DevOps
```
